scenario/state.py

```diff
@@ -1217,6 +1317,9 @@ class Event(_DCBase):
     # if this is a workload (container) event, the container it refers to
     container: Optional[Container] = None
 
+    # if this is a Pebble notice event, the notice it refers to
+    notice: Optional[PebbleNotice] = None
```

Comment:
I like `notice` because it's short, but @benhoyt @PietroPasotti do you think this really ought to be `pebble_notice` in case there are other things called "[x]notice" in the future? It felt safe enough to have `Container.notices` - hopefully there are never other kinds of notices inside of a container - but I'm not sure about at the more top-level namespace that `Event` sits in.

Reply:
I think `Container.notices` is fine because that's under container, but this one I'm on the fence about. However, I don't think there'll be other kinds of "[x]notice" in the future -- we'll avoid that name because it's a first-class Pebble concept. So I lean towards keeping it short `Event.notice`.

We could even change `scenario.PebbleNotice` to `scenario.Notice`. That one lacks any other context/namespace apart from "scenario", but I still think it's highly unlikely for there to ever be other scenario.FooNotice types, and even if there are, we can always use a prefix for that if that ever happens.